LanguageFeatures/Subtyping/dynamic/generated/records_arguments_binding_A01_t02.dart

```diff
@@ -0,0 +1,129 @@
+/// of T0 can be used as an argument of type T1. Test superclass members
+/// @author sgrekhov@unipro.ru
+///
+/// This test is generated from records_A01.dart and 
+/// arguments_binding_x02.dart.
+/// Don't modify it. If you want to change this test, change one of the files 
+/// above and then run generator.dart to regenerate the tests.
+
+
+import '../../utils/common.dart';
+
+// SharedOptions=--enable-experiment=records
+
+(int x, double y, String) t0Instance = (42, 3.14, "");
+(num, num z, String?) t1Instance = (0, 0, null);
+
+const t1Default = (0, 0, null);
+
+
+
+
+class ArgumentsBindingSuper1_t02 {
+  (num, num, String?) m;
+
+  ArgumentsBindingSuper1_t02((num, num, String?) value): m = value {}
+  ArgumentsBindingSuper1_t02.named((num, num, String?) value, {(num, num, String?) val2 = t1Default}): m = value {}
+  ArgumentsBindingSuper1_t02.positional((num, num, String?) value, [(num, num, String?) val2 = t1Default]): m = value {}
+  ArgumentsBindingSuper1_t02.short(this.m);
+
+  void superTest((num, num, String?) val) {}
+  void superTestPositioned((num, num, String?) val, [(num, num, String?) val2 = t1Default]) {}
+  void superTestNamed((num, num, String?) val, {(num, num, String?) val2 = t1Default}) {}
+  (num, num, String?) get superGetter => m;
+  void set superSetter((num, num, String?) val) {}
+}
+
+class ArgumentsBinding1_t02 extends ArgumentsBindingSuper1_t02 {
+  ArgumentsBinding1_t02(dynamic t1) : super(t1) {}
+  ArgumentsBinding1_t02.c2(dynamic t1, dynamic t2) : super.named(t1, val2: t2) {}
+  ArgumentsBinding1_t02.c3(dynamic t1) : super.positional(t1) {}
+  ArgumentsBinding1_t02.c4(dynamic t1, dynamic t2) : super.positional(t1, t2) {}
+  ArgumentsBinding1_t02.c5(dynamic t1) : super.short(t1) {}
+
+  test(dynamic t1, dynamic t2) {
+    superTest(t1);
+    superTestPositioned(t1);
+    superTestPositioned(t2, t1);
+    superTestNamed(t1);
+    superTestNamed(t2, val2: t1);
+    superSetter = t1;
+    m = t1;
+    superGetter;
+  }
+}
+
+class ArgumentsBindingSuper2_t02<X> {
+  X m;
+
+  ArgumentsBindingSuper2_t02(X value): m = value {}
+  ArgumentsBindingSuper2_t02.named(X value, {required X val2}):m = value {}
+  ArgumentsBindingSuper2_t02.short(this.m);
+
+  void superTest(X val) {}
+  void superTestNamed(X val, {required X val2}) {}
+  X get superGetter => m;
+  void set superSetter(X val) {}
+}
+
+class ArgumentsBinding2_t02<X> extends ArgumentsBindingSuper2_t02<X> {
+  ArgumentsBinding2_t02(X t1) : super(t1) {}
+  ArgumentsBinding2_t02.c2(dynamic t1, dynamic t2) : super.named(t1, val2: t2) {}
+  ArgumentsBinding2_t02.c5(dynamic t1) : super.short(t1) {}
+
+  test(X t1, X t2) {
+    superTest(t1);
+    superTestNamed(t2, val2: t1);
+    superSetter = t1;
+    m = t1;
+    superGetter;
+  }
+}
+
+main() {
+  ArgumentsBinding1_t02 c1 = new ArgumentsBinding1_t02(forgetType(t0Instance));
+  c1 = new ArgumentsBinding1_t02.c2(t1Instance, forgetType(t0Instance));
```

Comment:
It looks like `forgetType` doesn't matter here, because we're passing `t0Instance` to a parameter of type `dynamic`, right? Same for several other call sites in this library.

Reply:
The situation is similar here, code generation might be able to be more precise, and the overall quality of the generated test will be higher.

Reply:
Agreed. I'll fix it separately, the change will affect a lot of other generated tests

Reply:
OK.